src/Hyperion.Tests/UnsafeDeserializationExclusionTests.cs

```diff
@@ -22,5 +24,64 @@ public void CantDeserializeANaughtyType()
                     serializer.Deserialize<DirectoryInfo>(stream));
             }
         }
+
+        internal class ClassA
+        { }
+        
+        internal class ClassB
+        { }
+        
+        internal class ClassC
+        { }
+        
+        [Fact]
+        public void TypeFilterShouldThrowOnNaughtyType()
+        {
+            var typeFilter = TypeFilterBuilder.Create()
+                .Include<ClassA>()
+                .Include<ClassB>()
+                .Build();
+
+            var options = SerializerOptions.Default
+                .WithTypeFilter(typeFilter);
+
+            var serializer = new Serializer(options);
+            
+            using (var stream = new MemoryStream())
+            {
+                serializer.Serialize(new ClassA(), stream);
+                stream.Position = 0;
+                Action act = () => serializer.Deserialize<ClassA>(stream);
+                act.Should().NotThrow();
+                
+                stream.Position = 0;
+                Action actObj = () => serializer.Deserialize<object>(stream);
+                actObj.Should().NotThrow();
```

Comment:
Would this be enough to satistfy the Akka.NET test requirement?

Reply:
Absolutely

Reply:
Should also have a version of that that throws too when bad input is sent

Reply:
There is, its at the end of the test

Reply:
LGTM